transformer_engine/pytorch/tensor/float8_tensor.py

```diff
@@ -635,9 +733,92 @@ def __torch_dispatch__(cls, func, types, args, kwargs=None):
             )
         else:
             pass
-
         return super().__torch_dispatch__(func, types, args, kwargs)
 
+    def fsdp_pre_all_gather(self, mesh):
+        """Functions FSDP2 calls before all-gather of the
+        weights for both forward and backward passes.
+        Args:
+            mesh (torch.distributed.DeviceMesh): DeviceMesh used by FSDP2
+            to shard the weights.
+
+        Returns:
+            shareded_tensors: Tuple[torch.Tensor, ...]: Tuple of tensors
+            that need to be all-gathered.(In this case uint8 data tensor)
+            metadata: Tuple[Any]: Metadata needed for reconstructing the
+            Float8Tensor after all-gather.
+        """
+        quantizer = self._quantizer.copy()
+        if isinstance(quantizer, Float8CurrentScalingQuantizer) and mesh is not None:
+            # When sharded weight is updated after reduce scattering the gradients in FSDP2,
+            # we need to do amax reduction across the mesh to make sure all weight shards are
+            # updated with same scale inverse. Setting the state below in the quantizer will make
+            # sure that updated Quantized weight tensor have same scale inverse across all shards.
+            quantizer.amax_reduction_group = mesh.get_group()
+            quantizer.with_amax_reduction = True
+        is_transpose_needed = not self._transpose_invalid and self._transpose is not None
+        quantizer.set_usage(rowwise=not is_transpose_needed, columnwise=is_transpose_needed)
```

Comment:
I don't think this is correct. I would expect the sharded weights to be unchanged between the forward and backward pass, but the transpose requirements are different in the forward and backward pass.

If we remove transpose logic from `fsdp_post_all_gather`, we can simply remove this logic:
```suggestion
```